tests/test_expression_files.py

```diff
@@ -89,33 +89,123 @@ def mock_expression_load(self, *args):
 class TestExpressionFiles(unittest.TestCase):
     def test_check_unique_cells(self):
         client_mock = MagicMock()
-        header = ["GENE", "foo", "foo2", "foo3"]
 
         client_mock["data_arrays"].find.return_value = [
             {"values": ["foo3", "foo4", "foo5"]},
             {"values": ["foo6", "foo7", "foo8"]},
         ]
-        with self.assertRaises(ValueError) as cm:
-            GeneExpression.check_unique_cells(header, ObjectId(), client_mock)
-        self.assertTrue("contains 1 cells" in str(cm.exception))
-        self.assertTrue("foo3" in str(cm.exception))
 
+        """Following tests are for matrices that aren't raw counts"""
+
+        with patch(
+            "expression_files.expression_files.GeneExpression.get_raw_count_study_file_ids",
+            side_effect=None,
+            return_value=[],
+        ), self.assertRaises(ValueError) as cm:
+            header = ["GENE", "foo", "foo2", "foo3"]
+            GeneExpression.check_unique_cells(
+                header, ObjectId(), ObjectId(), client_mock
+            )
+            self.assertTrue("contains 1 cells" in str(cm.exception))
+            self.assertTrue("foo3" in str(cm.exception))
+
+        # Duplicate header but duplicate values don't appear in same order
         header = ["GENE", "foo", "foo3", "foo2", "foo6"]
-        with self.assertRaises(ValueError) as cm:
-            GeneExpression.check_unique_cells(header, ObjectId(), client_mock)
+        with patch(
+            "expression_files.expression_files.GeneExpression.get_raw_count_study_file_ids",
+            side_effect=None,
+            return_value=None,
+        ), self.assertRaises(ValueError) as cm:
+            GeneExpression.check_unique_cells(
+                header, ObjectId(), ObjectId(), client_mock
+            )
         self.assertTrue("contains 2 cells" in str(cm.exception))
         self.assertTrue("foo3" in str(cm.exception))
         self.assertTrue("foo6" in str(cm.exception))
 
-        header = ["GENE", "foo", "foo2"]
-        self.assertTrue(
-            GeneExpression.check_unique_cells(header, ObjectId(), client_mock)
-        )
+        with patch(
+            "expression_files.expression_files.GeneExpression.get_raw_count_study_file_ids",
+            side_effect=None,
+            return_value=None,
+        ):
+            header = ["GENE", "foo", "foo2"]
+            self.assertTrue(
+                GeneExpression.check_unique_cells(
+                    header, ObjectId(), ObjectId(), client_mock
+                )
+            )
+
+        """Following tests are for matrices that are raw counts"""
```

Comment:
Conventional single-line comment would be better here:

```suggestion
        # Following tests are for matrices that are raw counts
```